### PR TITLE
Digital Credentials: Framework/Setup

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#13385 & /bcgov/entity#13387

*Context:* 
* This is the start of a pilot project, through which businesses (specified Benefit Companies to start) will be able to create verified digital credentials and associate them to their business record (or somewhere, not 100% of how the relationship is made yet)

*Description of changes:*
* Implements routes and protections for Digital Credentials Flow
* New feature flag for Digital Credentials Pilot
* DC Views, placeholder components
* Simple Generic Stepper
* Minor unit test updates

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Review by vysakh-menon-aot — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "business-filings-ui",
-  "version": "5.2.21",
+  "version": "5.3.0",
```
Comment (severinbeauvais) [inline package.json:3]: Def go up a minor or major version. I think we're already at 5.3.x.

File: src/components/DigitalCredentials/ChooseCredentials.vue [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<template>
+  <div class="mt-8">CHOOSE PLACEHOLDER</div>
```
Comment (severinbeauvais) [inline src/components/DigitalCredentials/ChooseCredentials.vue:2]: I recommend that the top-level element have an id. Same with all new components below.

File: src/mixins/allowable-actions-mixin.ts [hunk 3]
```diff
@@ -82,6 +83,14 @@ export default class AllowableActionsMixin extends Vue {
         )
       }
 
+      case AllowableActions.VIEW_ADD_DIGITAL_CREDENTIALS: {
+        // Pilot is targeting specific Benefit Companies: Handled on LD Server Side
+        const isFeatureEnabled = !!getFeatureFlag('enable-digital-credentials')
+        const isNotaDcRoute = !(this.$route.matched.some(route => route.name === Routes.DIGITAL_CREDENTIALS))
+
+        return isFeatureEnabled && isNotaDcRoute && this.isGoodStanding && this.isBComp && !this.isRoleStaff
```
Comment (severinbeauvais) [inline src/mixins/allowable-actions-mixin.ts:91]: All parentheses around this please.

File: src/store/getters.ts [hunk 4]
```diff
@@ -241,6 +242,36 @@ export default {
...
+      },
+      {
+        id: 'step-3-btn',
+        step: 3,
+        icon: 'mdi-domain',
+        text: 'Scan with\n Digital Wallet',
+        to: `/digital-credentials/scan-credentials`,
+        component: ScanCredentials
+      }
+    ]
```
Comment (severinbeauvais) [inline src/store/getters.ts:274]: Since these have nothing to do with store state, they shouldn't be in getters.ts.

Instead, I recommend a mixin.

Comment (severinbeauvais) [reply src/store/getters.ts:274]: Or in the component itself if that's the only user of this getter?

**Review by severinbeauvais — APPROVED** [review]
Nicely written and laid out code, clean looking, good insertion into existing architecture. LGTM!

File: package.json (hunk 1 above)
Comment (cameron-freshworks) [reply package.json:3]: Copy that, thanks!

File: src/components/DigitalCredentials/ChooseCredentials.vue (hunk 2 above)
Comment (cameron-freshworks) [reply src/components/DigitalCredentials/ChooseCredentials.vue:2]: 100%, will add them where missing 👍

File: src/mixins/allowable-actions-mixin.ts (hunk 3 above)
Comment (cameron-freshworks) [reply src/mixins/allowable-actions-mixin.ts:91]: You bet, updated 👍

File: src/store/getters.ts (hunk 4 above)
Comment (cameron-freshworks) [reply src/store/getters.ts:274]: In this case, it would make sense in the parent component. Will update!

**Comment by cameron-freshworks** [comment]
> Nicely written and laid out code, clean looking, good insertion into existing architecture. LGTM!

Thanks sir, have a great holiday and thank you for the review :)

**Pull request merged**